src/main/java/kitchenpos/domain/OrderTable.java

```diff
@@ -1,40 +1,114 @@
 package kitchenpos.domain;
 
+import static java.util.Objects.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+
+@Entity
 public class OrderTable {
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Id
     private Long id;
-    private Long tableGroupId;
-    private int numberOfGuests;
+
+    @ManyToOne
+    @JoinColumn(name = "table_group_id")
+    private TableGroup tableGroup;
+
+    @Embedded
+    private NumberOfGuests numberOfGuests;
+
+    @OneToMany(mappedBy = "orderTable")
+    private List<Order> orders = new ArrayList<>();
+
     private boolean empty;
 
-    public Long getId() {
-        return id;
+    protected OrderTable() {}
+
+    public OrderTable(int numberOfGuests, boolean empty) {
+        this(numberOfGuests, empty, new ArrayList<>());
     }
 
-    public void setId(final Long id) {
-        this.id = id;
+    OrderTable(int numberOfGuests, boolean empty, List<Order> orders) {
+        this.numberOfGuests = NumberOfGuests.valueOf(numberOfGuests);
+        this.empty = empty;
+        this.orders = orders;
     }
 
-    public Long getTableGroupId() {
-        return tableGroupId;
+    public Long getId() {
+        return id;
     }
 
-    public void setTableGroupId(final Long tableGroupId) {
-        this.tableGroupId = tableGroupId;
+    public void ungrouped() {
+        validateCompletedOrders("조리상태이거나 식사상태인 주문이 있는 주문테이블은 그룹해제를 할 수 없습니다.");
+        this.tableGroup = null;
     }
 
     public int getNumberOfGuests() {
-        return numberOfGuests;
+        return numberOfGuests.getNumberOfGuests();
     }
 
-    public void setNumberOfGuests(final int numberOfGuests) {
-        this.numberOfGuests = numberOfGuests;
+    public void changeNumberOfGuests(final int numberOfGuests) {
+        if (isEmpty()) {
+            throw new IllegalArgumentException("빈 테이블은 방문 손님 수를 수정할 수 없습니다.");
+        }
+        this.numberOfGuests = NumberOfGuests.valueOf(numberOfGuests);
     }
 
     public boolean isEmpty() {
         return empty;
     }
 
-    public void setEmpty(final boolean empty) {
-        this.empty = empty;
+    public boolean isNotEmpty() {
+        return !empty;
+    }
+
+    public TableGroup getTableGroup() {
+        return tableGroup;
+    }
+
+    public void emptyOn() {
+        validate();
+        this.empty = true;
+    }
+
+    public void emptyOff() {
+        validate();
+        this.empty = false;
+    }
+
+    boolean isGrouped() {
+        return nonNull(tableGroup);
+    }
+
+    void toGroup(TableGroup tableGroup) {
+        this.tableGroup = tableGroup;
+    }
```

Comment:
아래처럼 비어있지 않은 상태로 지정해주는게 어떨까요?
```suggestion
    void toGroup(TableGroup tableGroup) {
        this.tableGroup = tableGroup;
        emptyOff();
    }
```